doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+Acknowledging an alarm updates alarm statistics and thereby applications like pmon can remove the particular alarm from status consideration.
+
+The alarm record in the ALARM table is marked with acknowledged field set to true. There is acknowledge-time field that indicates when that alarm is acknowledged.
+
+```
+sonic# alarm unacknowledge <seq-id-of-raised-alarm>
+```
+An operator can un-acknolwedge a previously acknowledged raised alarm. 
+Un-acknowledging an alarm updates alarm statistics and thereby applications like pmon can take the particular alarm into status consideration.
+
+The alarm record in the ALARM table is marked with acknowledged field set to false.
+There is acknowledge-time field that indicates when that alarm is un-acknowledged.
+
+```
+sonic# event profile <profile-name>
+```
+The command takes name of specified file, validates it for its syntax and values; merges it with its internal static map of events *static_event_map*.
+
+```
+sonic# clear event history
+```
+This command clears all the records in the event history table. All the event stats are cleared. 
+The command will not affect alarm table or alarm statistics.
+Eventd generates an event informing that event history table is cleared. 
+
+#### 3.3.2.2 Configuration Commands
+```
+sonic(config)# logging server <ip> [log|event] 
+```
+Note: The 'logging server' command is an existing, already supported command. 
+It is only enhanced to take either 'log' or 'event' to indicate either native syslog messages or syslog messages corresponding to events alone are sent to the remote host.
+Support with VRF/source-interface and configuring remote-port are all backward comaptible and will be applicable to either 'log' or 'event' options.
+
+#### 3.3.2.3 Show Commands
+```
+sonic# show event profile 
+--------------------------
+Active Event Profile
+--------------------------
+myProfile.json
+--------------------------
+Available Event Profiles
+--------------------------
+default.json
+myProfile.json
+userProfile.json
+
+sonic# show event [ details | summary | severity <sev> | start <from-ts> end <to-ts> | recent <5min|60min|24hr> | id <seq-id> | from <seq-id> to <seq-id> ] 
+
+'show event' commands would display all the records in EVENT table.
+
+sonic# show event 
+----------------------------------------------------------------------------------------------------------------------------
+Id           Action          Severity   Name                           Timestamp                   Description                                                  
+----------------------------------------------------------------------------------------------------------------------------
+1            -               WARNING    CUSTOM_EVPROFILE_CHANGE        2021-05-19T21:38:27.455Z    :- handle_custom_evprofile: Custom Event Profile x.json is applied.
```

Comment:
IMO Description start with :- is very strange. Why do we have it?

Reply:
Will change it.